src/main/java/uk/gov/ch/developer/docs/controller/developer/SignInController.java

```diff
@@ -0,0 +1,231 @@
+    private static final String secret = reader.getMandatoryString("COOKIE_SECRET");
+
+    protected Logger LOGGER = LoggerFactory.getLogger("docs.developer.ch.gov.uk");
+
+    @GetMapping
+    public void getSignIn(ServletRequest request, ServletResponse response) throws IOException {
+
+        Session chSession =
+                (Session) request.getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
+
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+
+        // Redirect for user authentication (no scope specified)
+        redirectForAuth(chSession, httpServletRequest, httpServletResponse, null, false);
+
+    }
+
+    /**
+     * Redirects to a URI for the user to authenticate themselves
+     * 
+     * @param session The user's session, retrieved from context
+     * @param request
+     * @param response
+     * @param companyNumber The company number a user is viewing
+     * @param force Boolean determining whether to force re-authentication
+     * @throws IOException
+     */
+    protected void redirectForAuth(Session session, HttpServletRequest request,
+            HttpServletResponse response, String companyNumber, boolean force) throws IOException {
+
+        // Find the original requested url
+        StringBuilder originalRequestUrl = new StringBuilder(request.getRequestURL().toString());
+        String queryString = request.getQueryString();
+        if (queryString != null)
+            originalRequestUrl.append("?").append(queryString);
+
+        // Set the scope
+        String scope = null;
+        if (companyNumber != null)
+            scope = OAUTH_COMPANY_SCOPE_PREFIX + companyNumber;
+
+        // Generate and store a nonce in the session
+        Session sessionToUpdate = session;
+
+        if (sessionToUpdate == null) {
+            sessionToUpdate = new SessionImpl();
+        }
+
+        String nonce = generateNonce();
+        sessionToUpdate.getData().put(SessionKeys.NONCE.getKey(), nonce);
+        
+//        //Store the CHS session
+//        chSession.store();
+
+        // Build oauth uri and redirect
+        String authoriseUri;
+        if (force) {
+            authoriseUri = createAuthoriseURIWithForceAndHint(originalRequestUrl.toString(), scope,
+                    nonce, getEmailFromSession(session));
+        } else {
+            authoriseUri = createAuthoriseURI(originalRequestUrl.toString(), scope, nonce);
+        }
+
+        response.sendRedirect(authoriseUri);
+    }
+
+    /**
+     * Retrieves email from a session if present, else returns an empty string
+     * 
+     * @param session User session from which to retrieve the email address
+     * @return email
+     */
+    private String getEmailFromSession(Session session) {
+
+        String email = "";
+
+        SignInInfo signInInfo = new SignInInfo();
+        if (session != null) {
```

Comment:
If session is null can't we just return null or an empty string? Because we know that the new SignInInfo has no user profile or content?